src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -67,108 +54,122 @@ func Setup(version string, buildDate string) *rotator.SizeRotator {
-	} else {
-		log.Debugf("Sending error logs to Loggly")
-		logglyClient = &logglyErrorWriter{loggly.New(logglyToken)}
-		errorOut = NonStopWriter(errorOut, logglyClient)
+		return
 	}
 
-	debugOut := timestamped(NonStopWriter(os.Stdout, file))
-	golog.SetOutputs(errorOut, debugOut)
-	return file
-}
-
-func Configure(cfg *config.Config) (err error) {
 	cfgMutex.Lock()
 	defer cfgMutex.Unlock()
 
-	if logglyClient == nil {
-		return fmt.Errorf("No loggly client to configure.")
+	if version == "" {
+		log.Error("No version configured, Loggly won't include version information")
+		return
 	}
 
-	if cfg.Addr == "" {
-		return fmt.Errorf("No proxy address provided.")
+	if buildDate == "" {
+		log.Error("No build date configured, Loggly won't include build date information")
+		return
 	}
 
-	if err = waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second); err != nil {
-		return fmt.Errorf("Proxy never came online at %s: %q", cfg.Addr, err)
-	}
+	if cfg.Addr != lastAddr {
+		lastAddr = cfg.Addr
 
-	var cli *http.Client
-	if cli, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr); err != nil {
-		return fmt.Errorf("Could not create proxy HTTP client %q.", err)
-	}
+		if cfg.Addr == "" {
+			log.Error("No known proxy, won't report to Loggly")
+			removeLoggly()
+			return
+		}
 
-	logglyClient.l.SetHTTPClient(cli)
+		err := waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second)
+		if err != nil {
+			log.Errorf("Proxy never came online at %v, not logging to Loggly", cfg.Addr)
+			removeLoggly()
+			return
+		}
+
+		var client *http.Client
+		client, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr)
+		if err != nil {
+			log.Errorf("Could not create proxied HTTP client, not logging to Loggly: %v", err)
+			removeLoggly()
+			return
+		}
+
+		log.Debugf("Sending error logs to Loggly via proxy at %v", cfg.Addr)
+
+		lang, _ := jibber_jabber.DetectLanguage()
+		logglyWriter := &logglyErrorWriter{
+			lang:            lang,
+			tz:              time.Local.String(),
+			versionToLoggly: fmt.Sprintf("%v (%v)", version, buildDate),
+			client:          loggly.New(logglyToken),
+		}
+		logglyWriter.client.SetHTTPClient(client)
+		addLoggly(logglyWriter)
+	}
+}
 
-	return nil
+func Close() error {
+	return logFile.Close()
 }
 
 // timestamped adds a timestamp to the beginning of log lines
 func timestamped(orig io.Writer) io.Writer {
 	return wfilter.LinePrepender(orig, func(w io.Writer) (int, error) {
-		return fmt.Fprintf(w, "%s - ", time.Now().In(time.UTC).Format(LogTimestampFormat))
+		return fmt.Fprintf(w, "%s - ", time.Now().In(time.UTC).Format(logTimestampFormat))
 	})
 }
 
-type logglyErrorWriter struct {
-	l *loggly.Client
+func addLoggly(logglyWriter io.Writer) {
+	golog.SetOutputs(io.MultiWriter(errorOut, logglyWriter), debugOut)
 }
 
-func (w logglyErrorWriter) Write(b []byte) (int, error) {
-	return writeToLoggly(w.l, "ERROR", string(b))
+func removeLoggly() {
+	golog.SetOutputs(errorOut, debugOut)
 }
```

Comment:
addLoggly and removeLoggly allow adding a logglyWriter to the base errorOut Writer.  This way, we can add remove Loggly whenever we're ready (e.g. once Configure() has been called) and we can change the Loggly settings simply by creating a new logglyWriter and adding that.